internal/controller/nificonnection_controller.go

```diff
@@ -757,6 +816,86 @@ func (r *NifiConnectionReconciler) DeleteConnection(ctx context.Context, clientC
 			}
 			return nil
 		}
+	} else if original.Spec.Destination.Type == v1alpha1.ComponentProcessGroup {
+		// Retrieve NifiResource information
+		destinationInstance, err := k8sutil.LookupNifiResource(r.Client, original.Spec.Destination.Name, original.Spec.Destination.Namespace)
+		if err != nil {
+			return err
+		}
+
+		// Check Resource is Process Group
+		if destinationInstance.Spec.Type != v1.ResourceProcessGroup {
+			return errors.New("source resource is expected to be Process Group, but actual reasource is not")
```

Comment:
```suggestion
			return errors.New(fmt.Sprintf("Destination resource %s in %s is expected to be Process Group",
				destinationInstance.Name, destinationInstance.Namespace))
```
To add more information, and make sure it's indicated that it's a `Destination` and not a `Source`.